### PR TITLE
Add Edit Decision Modal

### LINKED ISSUE #3710: Decisions: Create Edit Decisions details modal (UI Only)
### Story

As a user, after reviewing the Decision details i would like to be able to edit the details.

### Description

Based upon the Decisions details modal, open the modal with the previously entered information, change confirmation button text.

This should be based upon #3708
### Design 

Figma:https://www.figma.com/file/oRaTwsD5BUH3FJNzoZdSCv/Discussions?node-id=1018%3A56046

Screenshot 2022-08-02 at 13 12 58

### PR DESCRIPTION
**Pull request by willm30**
## Description

This PR wires up the "Edit" button on the Decisions Preview page with the `NewDecisionDialog` modal.

To test: 
* Go to /decisions/preview and click "Edit". Form content should be prefilled and submit button text should say "Update".
* Then to be sure go back to /decisions and click "New Decision". Form should be empty and button should say "Preview". 

**Changes** 🏗

* "Edit" onClick now opens `NewDecisionDialog`
* NewDecisionDialog takes optional "Title" and "Content" props so it can display the data entered previously when opened from Preview page

## Screenshot 

When being edited, if the form values don't change, the form will be disabled (because it wont be dirty), which I think is the correct behaviour.

editModal

Resolves #3710

### CODE REVIEW

File: src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx [hunk 1]
```diff
@@ -36,11 +38,12 @@ const MSG = defineMessages({
 const decisionData = {
   title: 'Should we build a Discord Bot?',
   description: `I think we should build a Discord bot that integrates with the Dapp and provides our community with greater transperency and also provides more convienience for us to be notified of things happening in our Colony.`,
+  /* Using an HTML string for the dialog content ensures the isDirty prop works as expected */
```
Comment (Julianahlk) [inline src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx:41]: How does this isDirty prop work? Is this related to Formik?

**Review by Julianahlk — APPROVED** [review]
Working very well for me 👍 Good job!

Screen Shot 2022-08-22 at 7 05 16 PM

Screen Shot 2022-08-22 at 7 04 53 PM

File: src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx (hunk 1 above)
Comment (arrenv) [reply src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx:41]: The prop should be just `dirty`, it is related to Formik. The purpose of it is to determine if the input value has been changed.

File: src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx [hunk 2]
```diff
@@ -36,11 +38,12 @@ const MSG = defineMessages({
...
+  htmlDescription: `<p>I think we should build a Discord bot that integrates with the Dapp and provides our community with greater transperency and also provides more convienience for us to be notified of things happening in our Colony.</p>`,
```
Comment (arrenv) [inline src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx:42]: Should this be HTML? One of the reasons for us picking TipTap was because it had a JSON input and output format. See here - https://tiptap.dev/guide/output

File: src/modules/dashboard/components/Dialogs/NewDecisionDialog/NewDecisionDialog.tsx [hunk 3]
```diff
@@ -18,16 +18,26 @@ const displayName = 'dashboard.NewDecisionDialog';
...
 
 const characterLimit = 4000;
 
-const NewDecisionDialog = ({ cancel, colony, ethDomainId }: Props) => {
+const NewDecisionDialog = ({
+  cancel,
+  colony,
+  ethDomainId,
+  decisionTitle = '',
+  content = '<p></p>',
```
Comment (arrenv) [inline src/modules/dashboard/components/Dialogs/NewDecisionDialog/NewDecisionDialog.tsx:39]: Does this need to be the default?

**Review by arrenv — APPROVED** [review]
Looks good @willm30, just some comments on using HTML for the Output

**Review by danbr — APPROVED** [review]
LGTM. Very good, well done.
It's very minor, but i would suggest renaming `NewDecisionDialog` to simply `DecisionDialog` or DecisionDetailsDialog` (or something else generic) as its now being used for creating and editing Decision details.

**Review by ArmandoGraterol — APPROVED** [review]
Looks good to me :ok_hand:

File: src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx (hunk 2 above)
Comment (willm30) [reply src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx:42]: @arrenv I'm only using an HTML string to keep track of the form state, so I can validate that the form is not empty. My cursory attempts just now to do that with an object were ineffective (presumably because two distinct objects with the same shape aren't equal to one another). 

The output format can still be JSON. Is that what you were querying?

Comment (arrenv) [reply src/modules/dashboard/components/DecisionPreview/DecisionPreview.tsx:42]: All good, yeah, that is what I was referring too.

File: src/modules/dashboard/components/Dialogs/NewDecisionDialog/NewDecisionDialog.tsx (hunk 3 above)
Comment (willm30) [reply src/modules/dashboard/components/Dialogs/NewDecisionDialog/NewDecisionDialog.tsx:39]: No, good spot.
